pkg/mediationcontainer/sdk_client_protocol.go

```diff
@@ -45,6 +67,11 @@ func (clientProtocol *SdkClientProtocol) handleClientProtocol(transport ITranspo
 	status = clientProtocol.HandleRegistration(transport)
 	if !status {
 		glog.Errorf("Failure during Registration, cannot receive server messages")
+		// panic here ... so Kubernetes can restart the probe pod
```

Comment:
Again, Kubernetes will not restart the pod when you do `os.Exit(1)`, only the container in the pod will be restarted. We should make it clear.